### PR TITLE
Generate SBOM and Provenance for released images

### PR DESCRIPTION
**Pull request by Polber**
PR adds SBOM and Provenance generation for released images, as well as Build time scanning for these images.

Notable changes are:
* Building images with Provenance (more info [here](https://cloud.google.com/build/docs/securing-builds/generate-validate-build-provenance).)
  * Images built with JIB (Java-only flex templates) are now packed as tar file instead of being pushed to GCR directly. These tar files are then sent to Cloud Build to be loaded (`docker load`) and pushed to GCR/AR - This takes advantage of the built-in Provenance support that Cloud Build provides for all images pushed to GCR/AR
  * Images built solely with Cloud Build (Python, Xlang and YAML templates) will still be built using the Kaniko executor, but will likewise be packaged as tar before being sent to next step where they are loaded (`docker load`) and pushed to GCR/AR.
* Generating SBOM
  * SBOM generation is invoked on the image(s) after being pushed to GCR/AR using the command
`gcloud artifacts sbom export --uri=gcr.io/...`.
More info about SBOM generation can be found [here](https://cloud.google.com/artifact-analysis/docs/generate-store-sboms#generate_an_sbom_file).
* Build-time scanning
  * ALL images will now be sent to Cloud Build as described in the Provenance section above. This not only allows us to leverage the Provenance generation, but also Build-time scanning. All the released images will be stored in a repository with continuous scanning, but this build-time scan allows critical vulnerabilities to be found before release, to hopefully avoid vulnerabilities earlier in the lifecycle of a released image.
  * This will use a separate Cloud Build job that runs after the image is staged. This job takes the released image as an argument and runs the scan against it.

All the above changes will be behind a maven argument `-DgenerateSBOM` which is disabled by default when Staging and Running a template (i.e. for ITs), but is enabled by default for Release.

### CODE REVIEW

**Review by liferoad — APPROVED** [review]

**Review by damccorm — COMMENTED** [review]
Just had one question, otherwise LGTM

File: plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesReleaseMojo.java [hunk 1]
```diff
@@ -116,6 +116,9 @@ public class TemplatesReleaseMojo extends TemplatesBaseMojo {
   @Parameter(defaultValue = "${unifiedWorker}", readonly = true, required = false)
   protected boolean unifiedWorker;
 
+  @Parameter(defaultValue = "true", property = "generateSBOM", readonly = true, required = false)
+  protected boolean generateSBOM;
```
[the remark this answers is not in the document]
Comment (Polber) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesReleaseMojo.java:120]: I'm not in a rush to get this in, let me re-run the workflow with it enabled and compare the diff

[the remark this answers is not in the document]
Comment (Polber) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesReleaseMojo.java:120]: Also, an argument for at least parameterizing the feature is that someone using the template plugin to stage a custom template or develop a new template would need access to Cloud Build which was not required before for Java-only templates

**Comment by Polber** [comment]
Test failure looks unrelated

File: plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesReleaseMojo.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Polber) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesReleaseMojo.java:120]: I might try running 1 more time, but the last run took right at 4 hours, meanwhile other PRs of mine recently ran in 2 hours

* https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/11275442726/job/31426291024?pr=1862
* https://github.com/GoogleCloudPlatform/DataflowTemplates/actions/runs/11282863188/job/31384565427?pr=1923

[the remark this answers is not in the document]
Comment (damccorm) [reply plugins/templates-maven-plugin/src/main/java/com/google/cloud/teleport/plugin/maven/TemplatesReleaseMojo.java:120]: That seems really high if its 4 hours - have you looked at the cloudbuild itself to generate the SBOM and checked how long that takes? 2 hours of SBOM generation is rough...

If that repros (or close to repros), then I agree, lets turn it off by default

**Pull request merged**